wsdl2java: -p has invalid character when when mapping multiple namespaces to Java packages

When using 

quarkus.cxf.codegen.wsdl2java.package-names[0]=test\=y
quarkus.cxf.codegen.wsdl2java.package-names[1]=jan\=x

or 
quarkus.cxf.codegen.wsdl2java.package-names=test\=y,jan\=y 

result in one -p parameter with two values separated by comma which is incorrect!. Should result in multiple -p parameter with one value.

For the record, the error message is 

```
[ERROR] Failed to execute goal io.quarkus:quarkus-maven-plugin:3.12.1:generate-code-tests (default) on project quarkus-cxf-integration-test-wsdl2java: Quarkus code generation phase has failed: InvocationTargetException: Could not run wsdl2Java -d /home/ppalaga/orgs/cxf/qcxf/integration-tests/wsdl2java/target/generated-test-sources/wsdl2java -p http://www.example.org/add=io.quarkiverse.cxf.wsdl2java.it.add,http://www.example.org/multiply=io.quarkiverse.cxf.wsdl2java.it.multiply /home/ppalaga/orgs/cxf/qcxf/integration-tests/wsdl2java/src/test/resources/wsdl/2ns.wsdl: org.apache.cxf.tools.common.toolspec.parser.BadUsageException: -p has invalid character!
```

--- a/extensions/core/deployment/src/main/java/io/quarkiverse/cxf/deployment/CxfBuildTimeConfig.java
+++ b/extensions/core/deployment/src/main/java/io/quarkiverse/cxf/deployment/CxfBuildTimeConfig.java
@@ -190,7 +190,7 @@ public interface CxfBuildTimeConfig {
          * @asciidoclet
          * @since 2.4.0
          */
-        @Wsdl2JavaParam(value = "-p", collection = Wsdl2JavaParamCollection.commaSeparated)
+        @Wsdl2JavaParam(value = "-p", collection = Wsdl2JavaParamCollection.multiParam)
         public Optional<List<String>> packageNames();
 
         /**

--- a/extensions/core/deployment/src/test/java/io/quarkiverse/cxf/deployment/codegen/Wsdl2JavaParamsTest.java
+++ b/extensions/core/deployment/src/test/java/io/quarkiverse/cxf/deployment/codegen/Wsdl2JavaParamsTest.java
@@ -46,7 +46,9 @@ public class Wsdl2JavaParamsTest {
         assertParams(
                 proxy(
                         "outputDirectory", Optional.of(QuarkusCxfClientTestUtil.maybeWinPath("foo/bar")),
-                        "packageNames", Optional.of(Arrays.asList("com.foo", "com.bar")),
+                        "packageNames",
+                        Optional.of(Arrays.asList("http://www.example.org/add=io.quarkiverse.cxf.wsdl2java.it.add",
+                                "http://www.example.org/multiply=io.quarkiverse.cxf.wsdl2java.it.multiply")),
                         "serviceName", Optional.of("HelloService"),
                         "bindings",
                         Optional.of(Arrays.asList(QuarkusCxfClientTestUtil.maybeWinPath("src/main/resources/b1.xml"),
@@ -69,7 +71,8 @@ public class Wsdl2JavaParamsTest {
                 "-nexclude", "http://foo.com",
                 "-nexclude", "http://bar.com",
                 "-mimeMethods", "mime1,mime2",
-                "-p", "com.foo,com.bar",
+                "-p", "http://www.example.org/add=io.quarkiverse.cxf.wsdl2java.it.add",
+                "-p", "http://www.example.org/multiply=io.quarkiverse.cxf.wsdl2java.it.multiply",
                 "-sn", "HelloService",
                 "-validate",
                 "-wsdlLocation", "my.wsdl",

--- a/integration-tests/wsdl2java/src/main/resources/application.properties
+++ b/integration-tests/wsdl2java/src/main/resources/application.properties
@@ -1,4 +1,9 @@
-quarkus.cxf.codegen.wsdl2java.includes = **.wsdl
-quarkus.cxf.codegen.wsdl2java.output-directory = target/generated-test-sources/wsdl2java-custom
-quarkus.cxf.codegen.wsdl2java.wsdl-location = classpath:wsdl/CalculatorService.wsdl
-quarkus.cxf.codegen.wsdl2java.xjc = bg,dv,javadoc,property-listener,ts,wsdlextension
+quarkus.cxf.codegen.wsdl2java.calculator.includes = wsdl/CalculatorService.wsdl
+quarkus.cxf.codegen.wsdl2java.calculator.output-directory = target/generated-test-sources/wsdl2java-custom
+quarkus.cxf.codegen.wsdl2java.calculator.wsdl-location = classpath:wsdl/CalculatorService.wsdl
+quarkus.cxf.codegen.wsdl2java.calculator.xjc = bg,dv,javadoc,property-listener,ts,wsdlextension
+
+quarkus.cxf.codegen.wsdl2java.2ns.includes = wsdl/multiple-package-names.wsdl
+quarkus.cxf.codegen.wsdl2java.2ns.package-names = http://www.example.org/math=io.quarkiverse.cxf.wsdl2java.it.math,
+  http://www.example.org/add=io.quarkiverse.cxf.wsdl2java.it.add,
+  http://www.example.org/multiply=io.quarkiverse.cxf.wsdl2java.it.multiply

--- a/integration-tests/wsdl2java/src/test/java/io/quarkiverse/cxf/wsdl2java/it/Wsdl2JavaTest.java
+++ b/integration-tests/wsdl2java/src/test/java/io/quarkiverse/cxf/wsdl2java/it/Wsdl2JavaTest.java
@@ -103,4 +103,17 @@ public class Wsdl2JavaTest {
                 .content(StandardCharsets.UTF_8).contains("implements ExtensibilityElement");
     }
 
+    @Test
+    void multiplePackageNames() {
+        /* Make sure that the java files for multiple-package-names.wsdl were generated in multiple packages */
+        Assertions.assertThat(Path.of("target/generated-test-sources/wsdl2java/org/example/add/AddRequest.java"))
+                .isRegularFile();
+        Assertions.assertThat(Path.of("target/generated-test-sources/wsdl2java/org/example/multiply/MultiplyRequest.java"))
+                .isRegularFile();
+        Assertions
+                .assertThat(Path
+                        .of("target/generated-test-sources/wsdl2java/io/quarkiverse/cxf/wsdl2java/it/math/MathService.java"))
+                .isRegularFile();
+    }
+
 }

--- a/integration-tests/wsdl2java/src/test/resources/wsdl/multiple-package-names.wsdl
+++ b/integration-tests/wsdl2java/src/test/resources/wsdl/multiple-package-names.wsdl
@@ -1,0 +1,103 @@
+<definitions xmlns="http://schemas.xmlsoap.org/wsdl/"
+             xmlns:soap="http://schemas.xmlsoap.org/wsdl/soap/"
+             xmlns:tns="http://www.example.org/math"
+             xmlns:add="http://www.example.org/add"
+             xmlns:multiply="http://www.example.org/multiply"
+             xmlns:xsd="http://www.w3.org/2001/XMLSchema"
+             targetNamespace="http://www.example.org/math">
+
+    <!-- Types section with two schemas in distinct namespaces -->
+    <types>
+        <xsd:schema targetNamespace="http://www.example.org/add">
+            <xsd:element name="AddRequest">
+                <xsd:complexType>
+                    <xsd:sequence>
+                        <xsd:element name="a" type="xsd:int"/>
+                        <xsd:element name="b" type="xsd:int"/>
+                    </xsd:sequence>
+                </xsd:complexType>
+            </xsd:element>
+            <xsd:element name="AddResponse">
+                <xsd:complexType>
+                    <xsd:sequence>
+                        <xsd:element name="result" type="xsd:int"/>
+                    </xsd:sequence>
+                </xsd:complexType>
+            </xsd:element>
+        </xsd:schema>
+
+        <xsd:schema targetNamespace="http://www.example.org/multiply">
+            <xsd:element name="MultiplyRequest">
+                <xsd:complexType>
+                    <xsd:sequence>
+                        <xsd:element name="a" type="xsd:int"/>
+                        <xsd:element name="b" type="xsd:int"/>
+                    </xsd:sequence>
+                </xsd:complexType>
+            </xsd:element>
+            <xsd:element name="MultiplyResponse">
+                <xsd:complexType>
+                    <xsd:sequence>
+                        <xsd:element name="result" type="xsd:int"/>
+                    </xsd:sequence>
+                </xsd:complexType>
+            </xsd:element>
+        </xsd:schema>
+    </types>
+
+    <!-- Messages -->
+    <message name="AddRequest">
+        <part name="parameters" element="add:AddRequest"/>
+    </message>
+    <message name="AddResponse">
+        <part name="parameters" element="add:AddResponse"/>
+    </message>
+    <message name="MultiplyRequest">
+        <part name="parameters" element="multiply:MultiplyRequest"/>
+    </message>
+    <message name="MultiplyResponse">
+        <part name="parameters" element="multiply:MultiplyResponse"/>
+    </message>
+
+    <!-- PortType -->
+    <portType name="MathPortType">
+        <operation name="Add">
+            <input message="tns:AddRequest"/>
+            <output message="tns:AddResponse"/>
+        </operation>
+        <operation name="Multiply">
+            <input message="tns:MultiplyRequest"/>
+            <output message="tns:MultiplyResponse"/>
+        </operation>
+    </portType>
+
+    <!-- Binding -->
+    <binding name="MathBinding" type="tns:MathPortType">
+        <soap:binding transport="http://schemas.xmlsoap.org/soap/http" style="document"/>
+        <operation name="Add">
+            <soap:operation soapAction="http://www.example.org/math/Add"/>
+            <input>
+                <soap:body use="literal"/>
+            </input>
+            <output>
+                <soap:body use="literal"/>
+            </output>
+        </operation>
+        <operation name="Multiply">
+            <soap:operation soapAction="http://www.example.org/math/Multiply"/>
+            <input>
+                <soap:body use="literal"/>
+            </input>
+            <output>
+                <soap:body use="literal"/>
+            </output>
+        </operation>
+    </binding>
+
+    <!-- Service -->
+    <service name="MathService">
+        <port name="MathPort" binding="tns:MathBinding">
+            <soap:address location="http://www.example.org/math"/>
+        </port>
+    </service>
+</definitions>

